src/main/java/de/tum/in/www1/artemis/web/rest/CourseResource.java

```diff
@@ -301,7 +305,7 @@ public ResponseEntity<OnlineCourseConfiguration> updateOnlineCourseConfiguration
     @PostMapping("courses/{courseId}/enroll")
     @EnforceAtLeastStudent
     public ResponseEntity<Set<String>> enrollInCourse(@PathVariable Long courseId) {
-        Course course = courseRepository.findWithEagerOrganizationsElseThrow(courseId);
+        Course course = courseRepository.findWithEagerOrganizationsAndCompetenciesAndLearningPathsElseThrow(courseId);
```

Comment:
Yo reload the course below, so you don't need to load them here as well. (Or you can also remove the db call below)
```suggestion
        Course course = courseRepository.findWithEagerOrganizationsElseThrow(courseId);
```

Reply:
You probably got confused by the Github view. This is the Method:
```
public ResponseEntity<Set<String>> enrollInCourse(@PathVariable Long courseId) {
        Course course = courseRepository.findWithEagerOrganizationsAndCompetenciesAndLearningPathsElseThrow(courseId);
        User user = userRepository.getUserWithGroupsAndAuthoritiesAndOrganizations();
        log.debug("REST request to enroll {} in Course {}", user.getName(), course.getTitle());
        courseService.enrollUserForCourseOrThrow(user, course);
        return ResponseEntity.ok(user.getGroups());
    }
```
I don't load the course twice.